md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java

```diff
@@ -175,6 +275,27 @@ RestWrapper delete(@PathVariable("cg") String configGroup, @PathVariable("key")
         return restWrapper;
     }
 
+    /**
+     * This method fetches all distinct record in GeneralConfig table depending on Config group.
+     *
+     * @return restWrapper with the list of instance of GeneralConfig table.
+     */
+
+    @RequestMapping( method = RequestMethod.GET)
+    @ResponseBody public
+    RestWrapper getDistinctGenerelConfig(Principal principal) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "principal". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)